app/controllers/budgets_controller.rb

```diff
@@ -1,46 +1,53 @@
 class BudgetsController < ApplicationController
   before_action :require_login
-
-  def index
-    @budgets = current_user.budgets
-  end
+  before_action :budget_exists, only: [:new, :create]
 
   def new
     @budget = Budget.new
   end
 
   def create
-    @budget = current_user.budgets.create(budget_params)
+    @budget = Budget.create(budget_params)
+    @budget.user_id = current_user.id
+    @budget.save
 
-    respond_with(@budget)
+    respond_with(@budget, location: -> { budgets_path })
   end
 
   def show
+    @budget = current_user.budget
   end
 
   def edit
+    @budget = current_user.budget
   end
 
   def update
-    budget.update_attributes(budget_params)
-
-    respond_with(budget)
+    if current_user.budget.update_attributes(budget_params)
```

Comment:
A suggestion to pull out a budget helper method... i.e.

```
private

def budget
  @budget ||= current_user.budget
end
helper_method :budget
```

Reply:
Duh, meant to go back and do this, thanks - done sha: 2149abb